pallets/index/src/lib.rs

```diff
@@ -0,0 +1,533 @@
+			Self::deposit_event(Event::WorkerAdd { worker });
+			Ok(())
+		}
+
+		#[pallet::weight(195_000_000)]
+		#[pallet::call_index(2)]
+		#[transactional]
+		pub fn deposit_task(
+			origin: OriginFor<T>,
+			asset: XcmAssetId,
+			amount: u128,
+			recipient: Vec<u8>,
+			worker: [u8; 32],
+			request_id: RequestId,
+			request: Vec<u8>,
+		) -> DispatchResult {
+			let sender: [u8; 32] = ensure_signed(origin)?.into();
+
+			// Ensure given worker was registered
+			ensure!(Workers::<T>::get(&worker), Error::<T>::WorkerNotSet);
+			// Check if foreign asset was registered on our chain
+			if &asset != &MultiLocation::here().into() {
+				let asset_location = match &asset {
+					Concrete(location) => Some(location),
+					_ => None,
+				}
+				.ok_or(Error::<T>::AssetNotFound)?;
+				ensure!(
+					T::AssetsRegistry::id(&asset_location).is_some(),
+					Error::<T>::AssetNotFound
+				);
+			}
+
+			// Check if record already exist
+			ensure!(
+				DepositRecords::<T>::get(&request_id).is_none(),
+				Error::<T>::RequestAlreadyExist
+			);
+
+			// Save record to corresponding worker task queue
+			let mut worker_task_queue = ActivedRequests::<T>::get(&worker);
+			worker_task_queue.push_back(request_id);
+			ActivedRequests::<T>::insert(&worker, &worker_task_queue);
+			// Save record data
+			let deposit_info = DepositInfo {
+				sender,
+				asset: asset.clone(),
+				amount,
+				recipient,
+				request,
+			};
+			DepositRecords::<T>::insert(&request_id, &deposit_info);
+
+			// Withdraw from sender account
+			T::AssetTransactor::withdraw_asset(
+				&(asset.clone(), amount).into(),
+				&Junction::AccountId32 {
+					network: NetworkId::Any,
+					id: sender.into(),
+				}
+				.into(),
+			)
+			.map_err(|_| Error::<T>::TransactFailed)?;
+			// Deposit into module account
+			let module_account: T::AccountId = MODULE_ID.into_account_truncating();
+			T::AssetTransactor::deposit_asset(
+				&(asset.clone(), amount).into(),
+				&Junction::AccountId32 {
+					network: NetworkId::Any,
+					id: module_account.into(),
+				}
+				.into(),
+			)
+			.map_err(|_| Error::<T>::TransactFailed)?;
+
+			Self::deposit_event(Event::NewRequest { deposit_info });
+			Ok(())
+		}
+
+		/// Claim the oldest task that saved in actived task queue
+		#[pallet::weight(195_000_000)]
+		#[pallet::call_index(3)]
+		#[transactional]
+		pub fn claim_task(origin: OriginFor<T>, request_id: RequestId) -> DispatchResult {
+			// Check origin, must be the worker
+			let worker: [u8; 32] = ensure_signed(origin)?.into();
+			ensure!(
+				Workers::<T>::get(&worker) == true,
+				Error::<T>::WorkerMismatch
+			);
+
+			let mut worker_task_queue = ActivedRequests::<T>::get(&worker);
+			// Check reqeust exist in actived task queue
+			ensure!(
+				worker_task_queue.contains(&request_id),
+				Error::<T>::NotFoundInTaskQueue
+			);
+			// Pop the oldest task from worker task queue
+			let _ = worker_task_queue
+				.pop_front()
+				.ok_or(Error::<T>::TaskQueueEmpty)?;
```

Comment:
I guess you meant to extract the `request_id` from the queue, instead of always popping the first element?

Reply:
Yes, fixed